doc/guides/zephyr_cmake_package.rst

```diff
@@ -382,6 +382,48 @@ Such a CMakeLists.txt could look as:
 
    project(my_first_app)
 
+.. _cmake_extension_package:
+
+Zephyr Extension CMake package
+******************************
+
+The Zephyr Extension CMake package provides a possibility for a Zephyr based system to control
```

Comment:
I think the naming here is awkward, I understand what a cmake package is, i.e. it lets an application find the zephyr base directory, but what does this extension do? what does it extend?
The description below does not tell me who is extending who and what is the scope of the extension.

Reply:
Maybe it will be useful to have a problem description first, i.e. describe the problem we are trying to solve then go into how all of this solves the problem.

Reply:
The basic problem is that our Zephyr modules are a bit fixed in when they are processed.

Of course users can modify Zephyr repo anyway they like, but this comes with the cost of keeping a patchset in synch.

This feature at least allows users to hook in early, that is before Kconfig, dts, toolchain, etc. to allow them to have their own code processed before anything in Zephyr project without directly patching Zephyr.

After TSC, I completely agree this feature must have a better naming.

Reply:
I have renamed everything into `Zephyr Build Configuration CMake package` and updated the docs to reflect that this functionality is in many ways comparable to a shared `.zephyrrc`.

The feature is a bit more powerful as it can do much more than `.zephyrrc`, but I think the comparison to `.zephyrrc` makes it much easier to understand the basics, and once users understand that, they can easier discover the rest.

Added an additional option here: 
https://github.com/zephyrproject-rtos/zephyr/pull/25575/files#diff-aef6d48923d8e3f151ae632679dc9c9f